src/main/java/org/folio/service/budget/BudgetService.java

```diff
@@ -45,8 +43,7 @@ public class BudgetService {
   public static final String TRANSACTION_IS_PRESENT_BUDGET_DELETE_ERROR = "transactionIsPresentBudgetDeleteError";
 
   public static final String SELECT_BUDGETS_BY_FY_AND_FUND_FOR_UPDATE = "SELECT jsonb FROM %s "
-    + "WHERE jsonb->>'fiscalYearId' = $1 AND jsonb->>'fundId' = $2 "
-    + "FOR UPDATE";
+    + "WHERE fiscalYearId::text = $1 AND fundId::text = $2 FOR UPDATE";
```

Comment:
and without adding ::text search is not working?

Reply:
Actually it works. I thought I might have to convert from the uuid type.

Reply:
in this case better to change all places without ::text convertions

Reply:
story is created to use that index: https://folio-org.atlassian.net/browse/MODFISTO-466